move/squid/sources/squid/object_procurement.move

```diff
@@ -0,0 +1,184 @@
+module squid::object_procurement;
+
+use std::ascii::{Self, String};
+use std::type_name;
+
+use sui::coin::Coin;
+use sui::bcs::{Self, BCS};
+
+use relayer_discovery::transaction::{Self, MoveCall};
+
+use squid::swap_info::SwapInfo;
+
+const SWAP_TYPE: u8 = 4;
```

Comment:
not a fan of partial declarations. define swap type enum in a separate type module and use it where needed?